client/src/main/java/org/eclipse/hono/client/impl/CommandConnectionImpl.java

```diff
@@ -95,12 +96,16 @@ protected void clearState() {
 
         final MessageConsumer messageConsumer = commandReceivers.get(Device.asAddress(tenantId, deviceId));
         if (messageConsumer != null) {
-            log.debug("reusing existing command consumer [tenant: {}, device-id: {}]", tenantId, deviceId);
-            return Future.succeededFuture(messageConsumer);
+            log.debug("cannot create concurrent command consumer [tenant: {}, device-id: {}]", tenantId, deviceId);
+            return Future.failedFuture(new ResourceConflictException("message consumer already in use"));
         } else {
             return createConsumer(
                     tenantId,
-                    () -> newCommandConsumer(tenantId, deviceId, commandConsumer, closeHandler));
+                    () -> newCommandConsumer(tenantId, deviceId, commandConsumer, closeHandler))
+               .map(consumer -> {
+                    commandReceivers.put(Device.asAddress(tenantId, deviceId), consumer);
```

Comment:
Is that in any way synchronized? What would happen if two threads would call into this method at the same time?